lib/sda/host_msg.go

```diff
@@ -77,13 +86,24 @@ func (t *Token) Id() TokenID {
 	defer tokenMutex.Unlock()
 	if t.cacheId == TokenID(uuid.Nil) {
 		url := network.NamespaceURL + "token/" + t.EntityListID.String() +
-			t.RoundID.String() + t.ProtoID.String() + t.TreeID.String() +
+			t.RoundID.String() + t.ServiceID.String() + t.ProtoID.String() + t.TreeID.String() +
 			t.TreeNodeID.String()
 		t.cacheId = TokenID(uuid.NewV5(uuid.NamespaceURL, url))
 	}
 	return t.cacheId
 }
 
+func (t *Token) Clone() *Token {
+	t2 := Token{}
```

Comment:
But that was when using pointers, no? Here we just use slices.

```
	tok := &sda.Token{EntityListID: sda.EntityListID(uuid.NewV4())}
	tok2 := *tok
	if tok.EntityListID != tok2.EntityListID {
		t.Fatal("t and t2 should be the same")
	}
	tok2.EntityListID = sda.EntityListID(uuid.NewV4())
	if tok.EntityListID == tok2.EntityListID {
		t.Fatal("t and t2 should differ now")
	}
```
passes. Or do I miss something?